src/compute/src/render/mod.rs

```diff
@@ -152,10 +152,11 @@ pub fn build_compute_dataflow<A: Allocate>(
 ) {
     // Mutually recursive view definitions require special handling.
     let recursive = dataflow.objects_to_build.iter().any(|object| {
-        use mz_expr::CollectionPlan;
-        let mut depends_on = BTreeSet::new();
-        object.plan.depends_on_into(&mut depends_on);
-        depends_on.into_iter().any(|id| id >= object.id)
+        if let Plan::LetRec { .. } = object.plan {
```

Comment:
maybe `matches!(object.plan, Plan::LetRec { .. })` instead?

Reply:
Idk tbh I'm morally opposed to macros when vanilla Rust would work. If someone wants to change this behind my back, maybe that is the better way. :D